src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this[0] !== null && this[0] !== undefined && this[0] !== '') {
+      result = `${this[0]}`;
+    }
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 1; i <= this.length - 1; i++) {
+      result += separator;
+
+      if (this[i] === null || this[i] === undefined || this[i] === '') {
+        continue;
+      } else {
+        result += this[i];
+      }
+    }
```

Comment:
You shoud add separator at the end in case when `i <= this.length - 1`
also you add to result only in case when `this[i] !== null && this[i] !== undefined`
```suggestion
    for (let i = 0; i < this.length; i++) {
      if (this[i] !== null && this[i] !== undefined) {
        // add to result 
      } 
      
      if (i < this.length - 1) {
        result += separator;
      }
    }
```